kartothek/io_components/write.py

```diff
@@ -246,3 +253,28 @@ def raise_if_dataset_exists(dataset_uuid, store):
                 )
     except KeyError:
         pass
+
+
+def check_dataset_creation_constraints(
+    factory, partition_on, secondary_indices, overwrite=False
+):
+    """Check if all required constraints for creating a dataset are met."""
+
+    if overwrite is not False:
+        warnings.warn(
+            "The keyword `overwrite` be removed in the next major release ",  # FIXME: keep or not?
```

Comment:
I think we should keep it. "Store" vs. "Update" have different semantics and for "store" I see very reasonable use cases in overwriting datasets

Reply:
- [ ] keep it